octodns/processor/arpa.py

```diff
@@ -56,7 +59,11 @@ def populate(self, zone, target=False, lenient=False):
         for arpa, fqdns in self._records.items():
             if arpa.endswith(f'.{zone_name}'):
                 name = arpa[:-n]
+                fqdns = list(set(fqdns))
```

Comment:
I think this will still have the problem of including the priority in the uniqueness. I believe the uniq-ifying would need to happen after line 64 where the `d[1]` bit is extracted, but that would potentially lose the sort order so this one is kind of interesting. 

May be easiest/cleanest to switch lines 63-65 to a more explicit setup like:

```python
seen = set()
ordered = sorted(fqdns)
fqdns = []
for _, fqdn in ordered:
    if fqdn in seen:
        continue
    fqdns.append(fqdn)
    seen.add(fqdn)
    if len(seen) >= self.max_auto_arpa:
        break
```

The trickiness of this definitely points towards the need for some specific testing around the behavior with duplicates fqdns & priorities, fqdns w/different priorities, etc. 

Might make sense to pull the logic out into a helper function:

```python
# Note: this takes a list of (priority, fqdn) tuples and returns the ordered and uniqified list of fqdns.
fqdns = self._order_and_unique_fqdns(fqdns, self.max_auto_arpa)
```

It would then be easy to call this method directly on an `AutoArpa` instance with all the various permutations without having to deal with creating records and all of the unrelated scaffolding.

Reply:
Great, thank you. I have added the suggestion.